### ISSUE TITLE
Optimize the BuildOwnersDictionary subroutine

### ISSUE THREAD

**Issue by tratteo** [Siamango/Solmango#9]
### Subject
The subroutine BuildOwnersDictionary builds a dictionary of owners for a certain collection.
### Problem
The problem is that Solana caps the requests at 10/s. For a collection of around 10 000 NFTs this implies that this methods would take almost 110 minutes to run.

### Possible solution   
Cache the list of owners at each iteration. 
When the iteration *i+1* runs, it will use the data cached at the iteration *i* and, by using the faster [GetOwnerMints](https://github.com/Siamango/Solmango/blob/525f5bc4dd547763c993359604e50e579d5190b4/SolmangoNET/Source/Wrappers.cs#L82) method, it will verify whether cached owners are still owners at the current interation. Since the probability of exchanging an NFT is surely lower than the probability of holding an NFT, the method should result optimized.

**Comment by Nickmadia** [comment]
Instead of building the dictionary as `<String,int>`,
i suggest that we build it as `<String, List <String>>` representing respectivily (`Owner`, List`<mintOwned>`), by doing so we can check the correctness of the cached dictionary by using the [GetOwnersMints](https://github.com/Siamango/Solmango/blob/525f5bc4dd547763c993359604e50e579d5190b4/SolmangoNET/Source/Wrappers.cs#L82) method and compare it to the cached dictionary so that the whole process becomes more efficient.

pseudo:
foreach(owner):// taken by the already built dictonary
mints = getOwnersMint;
if( owner has same mints): pass + remove mints from the list of the exchanged mints(at start this list represent the whole mint list)
else: update dic + update exchanged mint list

**Comment by tratteo** [comment]
> Instead of building the dictionary as `<String,int>`, i suggest that we build it as `<String, List <String>>` representing respectivily (`Owner`, List`<mintOwned>`), by doing so we can check the correctness of the cached dictionary by using the faster [GetOwnersMints](https://github.com/Siamango/Solmango/blob/525f5bc4dd547763c993359604e50e579d5190b4/SolmangoNET/Source/Wrappers.cs#L82) method and compare it to the cached dictionary so that the whole process becomes more efficient.
> 
> pseudo:
> 
> ```
> foreach(owner):// taken by the already built dictonary
> mints = getOwnersMint;
> if( owner has same mints): pass + remove mints from the list of the exchanged mints(at start this list represent the whole mint list)
> else: update dic + update exchanged mint list
> ```

That is right, however, since we are **forced** to execute the async routine `GetOwnerMints` anyway, instead of caching a dictionary we can just cache a list of the latest owners. Each iteration we call `GetOwnerMints` and, from that point on, we can use the solution you proposed.

**Comment by Nickmadia** [comment]
> > Instead of building the dictionary as `<String,int>`, i suggest that we build it as `<String, List <String>>` representing respectivily (`Owner`, List`<mintOwned>`), by doing so we can check the correctness of the cached dictionary by using the faster [GetOwnersMints](https://github.com/Siamango/Solmango/blob/525f5bc4dd547763c993359604e50e579d5190b4/SolmangoNET/Source/Wrappers.cs#L82) method and compare it to the cached dictionary so that the whole process becomes more efficient.
> > pseudo:
> > ```
> > foreach(owner):// taken by the already built dictonary
> > mints = getOwnersMint;
> > if( owner has same mints): pass + remove mints from the list of the exchanged mints(at start this list represent the whole mint list)
> > else: update dic + update exchanged mint list
> > ```
> 
> That is right, however, since we are **forced** to execute the async routine `GetOwnerMints` anyway, instead of caching a dictionary we can just cache a list of the latest owners. Each iteration we call `GetOwnerMints` for each cached owner and, from that point on, we can use the solution you proposed.

ok right, of course i forgot to say that after this routine we **must** still use the `BuildOwnersDictionary` method on the exchangedMints list since we have to find the new owners.
